### PR TITLE
msm8916 support: paella

### PR DESCRIPTION
**Pull request by provokateurin**
Panels are broken on msm8916 and need to be loaded as module. I had to implement that first. Also the mdev setup was not correct and gave me some problems, but that is now fixed too.

Cc @minecrell @calebccff

### CODE REVIEW

**Comment by jld3103** [comment]
IMG_20210711_154739

File: Makefile [hunk 1]
```diff
@@ -104,14 +104,14 @@ splash/%.ppm.gz: splash/%.ppm
...
 	
-initramfs-%.cpio: initramfs/bin/busybox initramfs/init initramfs/init_functions.sh splash/%.ppm.gz splash/%-error.ppm.gz
+initramfs-%.cpio: initramfs/bin/busybox initramfs/init initramfs/init_functions.sh splash/%.ppm.gz splash/%-error.ppm.gz kernel-%.gz-dtb
 	@echo "CPIO  $@"
 	@rm -rf initramfs-$*
 	@cp -r initramfs initramfs-$*
 	@cp src/info-$*.sh initramfs-$*/info.sh
 	@cp splash/$*.ppm.gz initramfs-$*/splash.ppm.gz
 	@cp splash/$*-error.ppm.gz initramfs-$*/error.ppm.gz
-	@cp src/info-$*.sh initramfs-$*/info.sh
```
Comment (jld3103) [inline Makefile:114]: this probably was a copy paste error because it's already copied 3 lines above

File: initramfs/init_functions.sh [hunk 2]
```diff
@@ -49,18 +49,34 @@ setup_usb_configfs() {
 	echo "rndis" > $CONFIGFS/g1/configs/c.1/strings/0x409/configuration \
 		|| echo "  Couldn't write configration name"
 
-	# Make sure the node for the eMMC exists
-	if [ -z "$(ls $EMMC)" ]; then
-		fatal_error "$EMMC could not be opened, possible eMMC defect"
+	if [[ $EMMC != "" ]]; then
```
Comment (calebccff) [inline initramfs/init_functions.sh:52]: This should be
```bash
if [ -n "$EMMC" ]; then
...
```

Comment (calebccff) [reply initramfs/init_functions.sh:52]: And likewise for the other similar IF blocks

**Comment by calebccff** [comment]
This is super awesome!

File: initramfs/init_functions.sh (hunk 2 above)
Comment (jld3103) [reply initramfs/init_functions.sh:52]: it was first, but it didn't work :woman_shrugging:

Comment (calebccff) [reply initramfs/init_functions.sh:52]: oh really? oops,,, maybe `[[ -n "$EMMC" ]]` ?

File: .gitignore [hunk 3]
```diff
@@ -21,5 +19,6 @@
 /*.tar.bz2
 /*.tar.xz
 /*.lst
+/*.zip
```
Comment (MartijnBraam) [inline .gitignore:22]: How'd you get zip files here?

File: src/deviceinfo_msm8916 [hunk 4]
```diff
@@ -0,0 +1,6 @@
+deviceinfo_flash_offset_base="0x80000000"
```
Comment (MartijnBraam) [inline src/deviceinfo_msm8916:1]: These offsets aren't used?

File: .gitignore (hunk 3 above)
Comment (jld3103) [reply .gitignore:22]: the archive of the kernel is a zip

File: src/deviceinfo_msm8916 (hunk 4 above)
Comment (jld3103) [reply src/deviceinfo_msm8916:1]: ah i just copied them from pmos. i can remove them

Comment (jld3103) [reply src/deviceinfo_msm8916:1]: ah sorry i misunderstood you. they are in use:
https://github.com/dreemurrs-embedded/Jumpdrive/blob/master/Makefile#L84

Comment (jld3103) [reply src/deviceinfo_msm8916:1]: the code already was in place for sdm845

File: initramfs/init_functions.sh (hunk 2 above)
Comment (jld3103) [reply initramfs/init_functions.sh:52]: both don't work. i'll just leave it like it is because that works

**Comment by jld3103** [comment]
@MartijnBraam this is done. still anything you would like to have changed?

File: Makefile [hunk 5]
```diff
@@ -104,14 +107,14 @@ splash/%.ppm.gz: splash/%.ppm
...
```
Comment (Danct12) [inline Makefile:114]: Why is this removed?

File: Makefile [hunk 6]
```diff
@@ -104,14 +107,14 @@ splash/%.ppm.gz: splash/%.ppm
 	@echo "GZ    $@"
 	@gzip < $< > $@
 	
-initramfs-%.cpio: initramfs/bin/busybox initramfs/init initramfs/init_functions.sh splash/%.ppm.gz splash/%-error.ppm.gz
+initramfs-%.cpio: initramfs/bin/busybox initramfs/init initramfs/init_functions.sh splash/%.ppm.gz splash/%-error.ppm.gz kernel-%.gz-dtb
```
Comment (Danct12) [inline Makefile:110]: This line change breaks build on non-dtb devices.

File: Makefile (hunk 5 above)
Comment (jld3103) [reply Makefile:114]: it's already copied three lines above

File: Makefile (hunk 6 above)
Comment (jld3103) [reply Makefile]: should be fixed now

**Comment by jld3103** [comment]
in a clean env everything worked except the librem5 u-boot build, but that failed because of something else

**Comment by jld3103** [comment]
@Danct12 @MartijnBraam anything missing?

File: initramfs/init_functions.sh (hunk 2 above)
Comment (calebccff) [inline initramfs/init_functions.sh:52]: This is great but should probably be in it's own PR

**Comment by calebccff** [comment]
Hey, a few notes:

* Drop 0bd0200c7426ad61f6f9acdb829a8b8b0f0bdc4e, it's not relevant to the PR and is just subjective pretifying, easier to get this merged without it.

* Drop 0bd0200c7426ad61f6f9acdb829a8b8b0f0bdc4e for the same reason, users aren't running this makefile.

* It is probably worth fixing the panel driver to work as a builtin, most likely some dependency isn't properly registered in DT and the probe order is wrong. Should be ok as long as mdev doesn't impact the pinephone.

* Please squash commits, maybe have one for the README, one to add paella and one to add module loading / mdev.

File: initramfs/init_functions.sh (hunk 2 above)
Comment (jld3103) [reply initramfs/init_functions.sh:52]: There is also functionality to get paella working at all. Maybe I can split it.

**Comment by jld3103** [comment]
The commit you reference is not pretyfying, but it's needed to build, because the name of the targets changed due to the modules. Try building for pinephone with the previous commit, it won't work.
As I'm not developing msm8916 I don't know what the problem with the builtin panel driver is and people like @minecrell know more.
Yeah I'll squash the commits.

**Comment by Minecrell** [comment]
> It is probably worth fixing the panel driver to work as a builtin, most likely some dependency isn't properly registered in DT and the probe order is wrong. Should be ok as long as mdev doesn't impact the pinephone.

I suspect this is some complicated clock driver issue which honestly is not worth investigating at the moment unless you have too much time.
